### PR TITLE
EL30 lookup improvement

### LINKED ISSUE #14175: Expression Language 3.0 value lookup performance improvement
Lookup performance can be improved for certain Expression Language 3.0 values.  By optimizing EL 3.0 context handling and classloader calls, the evaluations of EL unscoped and static expressions perform better with this fix.  This is a port of Apache bug fix 62453.  Also includes Apache bug 63781 to support multiple java levels correctly.

### PR DESCRIPTION
**Pull request by epj**
fixes #14175 
By optimizing EL 3.0 context handling and classloader calls, the evaluations of EL unscoped and static expressions perform
better with this fix.  This is a port of Apache bug fix 62453.  Also includes 63781 to support newer java levels.

### CODE REVIEW

File: dev/com.ibm.websphere.javaee.el.3.0/src/javax/el/ImportHandler.java [hunk 1]
```diff
@@ -15,24 +15,267 @@
  * limitations under the License.
  */
 // wtlucy: Apache bug 57142 patch pulled in - file rev 1644523
+// epj: backfit EL 4.0 perf improvement
```
Comment (pnicolucci) [inline dev/com.ibm.websphere.javaee.el.3.0/src/javax/el/ImportHandler.java:18]: If we have a tomcat bug we're pulling in would be good to detail that here as well as Bill did in his edit above yours.

Comment (epj) [reply dev/com.ibm.websphere.javaee.el.3.0/src/javax/el/ImportHandler.java]: updated comment to mention apache bug

**Comment by volosied** [comment]
I see this also ports https://github.com/apache/tomcat/commit/45a65b9467371ffd9327d5d9485dcc55781f2296#diff-1b091e98ed797c9bf793ba4d987f0575a42335fc4c818f9f857b7e43b67670fc

Util#getMethod is updated to use the  JreCompat

**Comment by epj** [comment]
@volosied Good catch.  Didn't realize the java lvl changes are actually different bug #. I'll add mention of it to the issue.

**Comment by volosied** [comment]
No problem. I also noticed that ReflectionUtil.java should be updated as well for 63781.

**Comment by epj** [comment]
ReflectionUtil is in a different package, but it does appear it should be included so the change 63781 is complete so I'll add it to this PR.

**Comment by epj** [comment]
#build
#spawn.fullfat.buckets=com.ibm.ws.jsp_fat,com.ibm.ws.jsp.2.3_fat,com.ibm.ws.jsp_fat_lWAS,com.ibm.ws.jsp_fat_jsp.2.3.toleration,com.ibm.ws.el.3.0_fat,com.ibm.ws.jsf.2.3_fat

**Review by pnicolucci — APPROVED** [review]

**Comment by epj** [comment]
#libbybot

**Pull request merged**